src/LinearMaps.jl

```diff
@@ -19,6 +19,23 @@ const MapOrMatrix{T} = Union{LinearMap{T},AbstractMatrix{T}}
 
 Base.eltype(::LinearMap{T}) where {T} = T
 
+abstract type MulStyle end
+
+struct FiveArg <: MulStyle end
+struct ThreeArg <: MulStyle end
+
+mulstyle(::Type{FiveArg}, ::Type{FiveArg}) = FiveArg
+mulstyle(::Type{ThreeArg}, ::Type{FiveArg}) = ThreeArg
+mulstyle(::Type{FiveArg}, ::Type{ThreeArg}) = ThreeArg
+mulstyle(::Type{ThreeArg}, ::Type{ThreeArg}) = ThreeArg
+mulstyle(::LinearMap) = ThreeArg # default
+if VERSION ≥ v"1.3.0-alpha.115"
+    mulstyle(::AbstractMatrix) = FiveArg
+else
+    mulstyle(::AbstractMatrix) = ThreeArg
+end
+mulstyle(A::LinearMap, As::LinearMap...) = mulstyle(mulstyle(A), mulstyle(As...))
+
```

Comment:
Looks much better, thanks. I'll need to contemplate a little about the exact difference, but getting rid of this `Type` stuff is already worth it.

Reply:
IIUC, then I don't need an `MS` type parameter in `LinearCombination`, right? The dispatch then doesn't go via its parameter, but via a dedicated argument in those functions that need it.

Reply:
I think I got the idea of a trait slightly wrong. Now, after rethinking your comment, I think I understood much better. I thought I'd save something by storing the mulstyle type as a type parameter. But I guess calling `MulStyle(A)` "over and over again" doesn't cost anything, because this can be determined at compile time. Which is good, because then I don't need to touch the type definition of `LinearCombination`s.